src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -133,24 +133,46 @@ private AdapterBinding getUserAdapterBinding(Property property, JsonbContext jso
 
     private PropertyCustomization introspectCustomization(Property property, JsonbContext jsonbContext) {
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
-        final CustomizationBuilder builder = new CustomizationBuilder();
+        final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
         if (introspector.isTransient(property)) {
             builder.setJsonbTransient(true);
             return builder.buildPropertyCustomization();
         }
         builder.setJsonReadName(introspector.getJsonbPropertyJsonReadName(property));
         builder.setJsonWriteName(introspector.getJsonbPropertyJsonWriteName(property));
-        builder.setNillable(classModel.getClassCustomization().isNillable()
-                || introspector.isPropertyNillable(property));
+        builder.setNillable(classModel.getClassCustomization().isNillable() || introspector.isPropertyNillable(property));
         builder.setAdapterInfo(getUserAdapterBinding(property, jsonbContext));
         builder.setSerializerBinding(getUserSerializerBinding(property, jsonbContext));
         builder.setDeserializerBinding(introspector.getDeserializerBinding(property));
         builder.setDateFormatter(introspector.getJsonbDateFormat(property));
-        builder.setNumberFormat(introspector.getJsonbNumberFormat(property));
+        introspectNumberFormatter(property, introspector, builder);
         return builder.buildPropertyCustomization();
     }
 
+    private void introspectNumberFormatter(Property property, AnnotationIntrospector introspector, PropertyCustomizationBuilder builder) {
+        /*
+         * If @JsonbNumberFormat is placed on getter implementation must use this format on serialization.
+         * If @JsonbNumberFormat is placed on setter implementation must use this format on deserialization.
+         * If @JsonbNumberFormat is placed on field implementation must use this format on serialization and deserialization.
+         */
+        Map<AnnotationTarget, JsonbNumberFormatter> jsonNumberFormatCategorized = introspector.getJsonNumberFormatter(property);
+        if(jsonNumberFormatCategorized.keySet().contains(AnnotationTarget.PROPERTY)) {
```

Comment:
How about overriding here: NumberFormat set on Class, on Field and on Setter. Class should be overriden by field (this works) but I am not sure if Field will get overriden by Setter for deserialization here. Or Class level overriden by Method level doesn't look to work here.